engine/consensus/matching/engine.go

```diff
@@ -343,31 +414,29 @@ func (e *Engine) onReceipt(originID flow.Identifier, receipt *flow.ExecutionRece
 	// finalizer when blocks are added to the chain, and the IncorporatedBlockID
 	// will be the ID of the first block on its fork that contains a receipt
 	// committing to this result.
-	err = e.storeIncorporatedResult(receipt, &log)
+	_, err = e.storeIncorporatedResult(receipt)
 	if err != nil {
-		if engine.IsDuplicatedEntryError(err) {
-			return nil
-		}
-		return fmt.Errorf("failed to store incorporated result: %w", err)
+		return false, fmt.Errorf("failed to store incorporated result: %w", err)
 	}
+
 	log.Info().Msg("execution result processed and stored")
 
-	return nil
+	return true, nil
 }
 
 // storeReceipt adds the receipt to the receipts mempool as well as to the persistent storage layer.
 // Return values:
+//  * bool to indicate whether the receipt is stored.
 // 	* `engine.DuplicatedEntryError` [sentinel error] if entry already present in mempool and we don't need to process this again
 //	* exception in case something went wrong
 // 	* nil in case of success
-func (e *Engine) storeReceipt(receipt *flow.ExecutionReceipt, head *flow.Header, log *zerolog.Logger) error {
+func (e *Engine) storeReceipt(receipt *flow.ExecutionReceipt, head *flow.Header) (bool, error) {
```

Comment:
Why? We have a dedicated sentinel error for this case. That is quite verbose comparing to using `bool`

Reply:
Sorry, I'm not clear about your question. Are you asking why to add a `bool`? it was to indicate whether the receipt was stored.